### PR TITLE
initial docs on adding a component to the index

### LINKED ISSUE #1731: [website] Write documentation for how to add a component to the community index
Document the process I used to add the ibm.com components to the community index, to act as the first draft of the documentation to go on the website.

### PR DESCRIPTION
**Pull request by shinytoyrobots**
Edited from the instructions in the Medium post.

I think we probably still need info on the dimensions / format of a component thumbnail image.

Will close #1731

### CODE REVIEW

**Comment by shinytoyrobots** [comment]
@janchild - Do you think this is in the right place in the IA (as a link within the "Community assets" dropdown?

We do need info on image dimensions/format - though I don't think that it's a blocker to releasing a first pass of instructions.

**Comment by aagonzales** [comment]
@shinytoyrobots @janchild I think it would make more since to move the contributing content to a tab inside the index page. It doesnt match the concept of "assets" its related to an asset but not an asset on its own.

**Comment by shinytoyrobots** [comment]
Good thinking, @aagonzales - I'll make that update.

**Comment by shinytoyrobots** [comment]
Last push has tabbed navigation example

**Comment by aagonzales** [comment]
@shinytoyrobots I'm still seeing "contribute a component" in the side nav.

**Comment by aagonzales** [comment]
This is also a small visual thing but can we add a line break after the code snippet block so the type in #5 isn't so close to the component? Or will that mess up the numbering? 

image

File: src/pages/community/how-to-contribute.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+title: Contributing a component
+description:
+  We want anyone to be able to add assets to the community component index.
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:4]:
```suggestion
  Anyone can add assets to the community component index.
```

File: src/pages/community/how-to-contribute.mdx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  Maintainers can submit a short pull request in GitHub to add components to the
+  index.
+
+tabs: ['Component index', 'How to contribute']
+---
+
+<PageDescription>
+
+We want anyone to be able to add assets to the community component index.
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:13]:
```suggestion
Anyone can add assets to the community component index.
```

File: src/pages/community/how-to-contribute.mdx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Maintainers can submit a short pull request in GitHub to add components to the
+index.
+
+</PageDescription>
+
+## Add a component using the GitHub UI
+
+1. In the
+   [index folder](https://github.com/carbon-design-system/carbon-website/tree/master/src/data/index)
+   of the carbon-website GitHub repo, open your team folder (or create one if
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:23]:
```suggestion
   of the carbon-website GitHub repo, open your team folder. Create a folder if
```

File: src/pages/community/how-to-contribute.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   your team doesn’t have on yet).
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:24]:
```suggestion
   your team doesn’t have one yet.
```

File: src/pages/community/how-to-contribute.mdx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+We want anyone to be able to add assets to the community component index.
+Maintainers can submit a short pull request in GitHub to add components to the
+index.
+
+</PageDescription>
+
+## Add a component using the GitHub UI
+
+1. In the
+   [index folder](https://github.com/carbon-design-system/carbon-website/tree/master/src/data/index)
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:22]:
```suggestion
   [index](https://github.com/carbon-design-system/carbon-website/tree/master/src/data/index) folder
```

File: src/pages/community/how-to-contribute.mdx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+   of the carbon-website GitHub repo, open your team folder (or create one if
+   your team doesn’t have on yet).
+2. On the next page, in the top right above the the table click `Add file` then
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:25]:
```suggestion
2. On the index page, click `Add file` (top right), then choose
```

File: src/pages/community/how-to-contribute.mdx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+   `Create new file`.
+3. Name your file `component-name.yml`.
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:27]:
```suggestion
3. Name your file `component-name.yml`, where _component-name_ is the name of your component. 
```

File: src/pages/community/how-to-contribute.mdx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+4. Copy and paste this template into the “Edit new file” section of the page and
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:28]:
```suggestion
4. Copy and paste the following template into the “Edit new file” section of the page and
```

File: src/pages/community/how-to-contribute.mdx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+   description: >-
+     short description of purpose and usage of component
+   framework: choose one of Vanilla | Angular | React | Vue
+   design_asset: choose one of Sketch | Axure | Adobe XD | Figma
+   platform: Web
+   aliases:
+     - fuzzy term for component (e.g. an alias for a "card" component might be "tile")
+   date_added: YYY-MM-DDT00:00:00.000Z
+   ```
+5. Once done, click the green `Create new file` button at the bottom.
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:46]: This button says **Commit new file** for me. Is it different for me, or is this a mistake?

File: src/pages/community/how-to-contribute.mdx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+6. After submitting the PR for the data, you’ll need to upload your image. Go
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:47]:
```suggestion
After submitting the PR for the data, you’ll need to upload your image. 

1. Go
```

File: src/pages/community/how-to-contribute.mdx [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+   back to your team folder in the
+   [index folder](https://github.com/carbon-design-system/carbon-website/tree/master/src/data/index).
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:49]:
```suggestion
   [index](https://github.com/carbon-design-system/carbon-website/tree/master/src/data/index) page.
```

File: src/pages/community/how-to-contribute.mdx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+   framework: choose one of Vanilla | Angular | React | Vue
+   design_asset: choose one of Sketch | Axure | Adobe XD | Figma
+   platform: Web
+   aliases:
+     - fuzzy term for component (e.g. an alias for a "card" component might be "tile")
+   date_added: YYY-MM-DDT00:00:00.000Z
+   ```
+5. Once done, click the green `Create new file` button at the bottom.
+6. After submitting the PR for the data, you’ll need to upload your image. Go
+   back to your team folder in the
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:48]:
```suggestion
   back to your team folder on the
```

File: src/pages/community/how-to-contribute.mdx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+   [index folder](https://github.com/carbon-design-system/carbon-website/tree/master/src/data/index).
+7. Inside your team folder click on the `images` folder.
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:50]:
```suggestion
7. Inside your team folder click the `images` folder.
```

File: src/pages/community/how-to-contribute.mdx [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+8. In the top right, click `Add file` then `Upload files` then upload your image
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:51]:
```suggestion
8. In the top right, click `Add file` then `Upload files` to upload your image
```

File: src/pages/community/how-to-contribute.mdx [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+   to that page then submit the PR by clicking the green `Commit changes` button
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:52]:
```suggestion
   to that page. 
   
    1. Submit the PR by clicking the green `Commit changes` button
```

File: src/pages/community/how-to-contribute.mdx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+   at the bottom. Note: name you image file with the same name as the component
+   (e.g. `component-name.png` to match `component-name.yml`).
+9. And you’re done! The Carbon team will review and merge the PR.
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:55]:
```suggestion
And you’re done! The Carbon team will review and merge the PR.
```

File: src/pages/community/how-to-contribute.mdx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+If you’re unsure about how to make a PR through GitHub then create an issue in
```
Comment (janchild) [inline src/pages/community/how-to-contribute.mdx:57]:
```suggestion
If you’re unsure about how to make a PR through GitHub, create an issue in
```

**Review by janchild — COMMENTED** [review]
Looks good. Just a few documentation smoothing tweaks in here.

**Comment by shinytoyrobots** [comment]
Latest push addresses all @janchild's tweaks and fixes that formatting issue for the code block, @aagonzales.

**Review by janchild — APPROVED** [review]

**Pull request merged**